Documentation/PlatformSupport.md

```diff
@@ -0,0 +1,69 @@
+
+# Platform Support
+
+Darwin platforms are binary and source stable. Linux platforms follow semantic versioning. Windows follows semantic versioning.
+
+### Windows concerns
+
+Trying to perfectly unify Windows with Darwin and Linux at the API level is ultimately a futile endeavor. However, in some specific cases, Windows does have directly corresponding syscalls or near-equivalents.
+
+Windows-specific details are open to reevaluation pending more Windows expertise for the project.
+
+
+## API to syscall mapping
+
+### `Errno`
+
+#### Instance computed properties
+
+| API                | Darwin     | Linux      | Windows    |
+|:-------------------|:-----------|:-----------|:-----------|
+| `description`      | `strerror` | `strerror` | `strerror` |
+| `debugDescription` | `strerror` | `strerror` | `strerror` |
+
+
+### `FileDescriptor`
+
+#### Instance methods
+
+| API                          | Darwin      | Linux       | Windows                                            |
+|:-----------------------------|:------------|:------------|:---------------------------------------------------|
+| `open`                       | `open`      | `open`      | `_wsopen_s(..., _SH_DENYNO, _S_IREAD | _S_IWRITE)` |
+| `close`                      | `close`     | `close`     | `_close`                                           |
+| `seek`                       | `lseek`     | `lseek`     | `_lseeki64`                                        |
+| `read`                       | `read`      | `read`      | `_read`                                            |
+| `read(fromAbsoluteOffset:)`  | `pread`     | `pread`     | *custom*                                           |
+| `write`                      | `write`     | `write`     | `_write`                                           |
+| `write(fromAbsoluteOffset:)` | `pwrite`    | `pwrite`    | *custom*                                           |
+| `duplicate`                  | `dup`       | `dup`       | `_dup`                                             |
+| `duplicate(as:)`             | `dup2`      | `dup2`      | `_dup2`                                            |
+| `resize`                     | `ftruncate` | `ftruncate` | *N/A*                                              |
```

Comment:
Hmm, `ftruncate` is possible to implement, though it requires a custom implementation.  I think that it was likely just missed when the API was added.

Reply:
I'm really sorry, but I have trouble understanding why this table needs to exist.

I beg y'all, *please* let's not pretend that Windows provides a syscall interface that matches Darwin or Linux. This package really isn't supposed to be platform-independent -- each platform ought to have its own set of APIs, tailored to its own peculiarities. Let each platform shine. Let each platform have its own native interfaces. Don't force one platform to needlessly (and badly) emulate the pointless minutiae of another! Embrace & expose the differences, as faithfully as possible.

Creating a set of portable APIs is a very desirable and extremely important goal; however, this package most certainly cannot fulfill that goal without losing its original reason for existence -- which was to enable native system programming without the complexities of C APIs naively imported into Swift.

The question shouldn't be whether it's possible to implement something on Windows that looks kind of like `ftruncate` from a distance -- rather, the question ought to be what can we do to faithfully expose the real `_wsopen_s` interface, or to make calling `CreateFileA/W` a pleasant experience in Swift.

Reply:
Actually, this set of interfaces are meant to be nearly identical (there are some differences in the sense that some require an extra parameter or such).  Windows does have a POSIX (to the letter) subsystem, so these interfaces are present.  Of course, being to the letter, optional interfaces may not present.  `resize` is not part of the POSIX standard that it implements, but the functionality to truncate the file is more or less just spelling differences.